src/core/service.c

```diff
@@ -2140,6 +2141,83 @@ _pure_ static bool service_can_reload(Unit *u) {
         return !!s->exec_command[SERVICE_EXEC_RELOAD];
 }
 
+static unsigned service_exec_command_index(Unit *u, ServiceExecCommand id, bool control) {
+        Service *s = SERVICE(u);
+        unsigned index = 0;
```

Comment:
nitpick, and sorry for noticing this earlier: there's a public glibc function "index()", I'd thus recommend not overloading the name locally here, but instead pick a different name, for example "idx" or just "i", or something like that. (old gcc even used to warn about this)